### PR TITLE
AMP-Ad Fast Fetch: Allow for specifying safeframe version via header

### PR DESCRIPTION
**Pull request by keithwrightbos**
Currently safeframe version used for rendering xdomain creatives is hardcoded however it can change as new versions are released.  Allow for specifying via ad response header "X-AmpSafeframeVersion".  Note that in the medium term, plan is to replace safeframe usage with AMP specific nameframe implementation.

Related to #9319 

/cc @ampproject/a4a

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -584,6 +586,12 @@ export class AmpA4A extends AMP.BaseElement {
           if (method && !isEnumValue(XORIGIN_MODE, method)) {
             dev().error('AMP-A4A', `cross-origin render mode header ${method}`);
           }
+          const safeframeVersion =
+            fetchResponse.headers.get(SAFEFRAME_VERSION_HEADER);
+          if (safeframeVersion && safeframeVersion != SAFEFRAME_VERSION) {
+            this.safeframeVersion_ = safeframeVersion;
```
Comment (jonkeller) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Functionally, code looks great. But variable names are a bit unclear. I'd suggest renaming:
SAFEFRAME_VERSION -> DEFAULT_SAFEFRAME_VERSION (or similar prefix to indicate this is a just a starting point)
const safeframeVersion -> safeframeVersionRequestedByHeader (or similar but less wordy)
this.safeframeVersion -> leave as-is

**Comment by bradfrizzell** [comment]
What was the benefit of switching from preconnect.url() to preconnect.preload() ?

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -71,13 +71,13 @@ const METADATA_STRING_NO_QUOTES =
...
 /** @type {string} */
-export const SAFEFRAME_VERSION = '1-0-8';
+export const DEFAULT_SAFEFRAME_VERSION = '1-0-8';
 
 /** @type {string} @visibleForTesting */
 export const RENDERING_TYPE_HEADER = 'X-AmpAdRender';
 
 /** @type {string} @visibleForTesting */
-export const SAFEFRAME_VERSION_HEADER = 'X-AmpSafeframeVersion';
+export const SAFEFRAME_VERSION_HEADER = 'X-AmpSafeFrameVersion';
```
Comment (jonkeller) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Nit: capitalization of the 'F' now inconsistent with line 337, 589, etc.

**Comment by taymonbeal** [comment]
Per [Google style](https://google.github.io/styleguide/jsguide.html#naming-camel-case-defined), it should be `SafeFrame` or `safeFrame`, not `Safeframe` or `safeframe`. Although, in the specific case of the header, you should match the capitalization of the header that the server sends, just in case.

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 3]
```diff
@@ -584,6 +586,13 @@ export class AmpA4A extends AMP.BaseElement {
           if (method && !isEnumValue(XORIGIN_MODE, method)) {
             dev().error('AMP-A4A', `cross-origin render mode header ${method}`);
           }
+          const safeframeVersionHeader =
+            fetchResponse.headers.get(SAFEFRAME_VERSION_HEADER);
+          if (safeframeVersionHeader &&
+              safeframeVersionHeader != DEFAULT_SAFEFRAME_VERSION) {
+            this.safeframeVersion_ = safeframeVersionHeader;
```
Comment (jridgewell) [inline extensions/amp-a4a/0.1/amp-a4a.js]: Do we need to worry about XSS here?

Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/amp-a4a.js]: I'll add a regexp check in another PR

**Pull request merged**